Feat/add age filter to projects

a-custodio opened the pull request:
This PR introduces a configurable time window for project analysis to prevent notifications for very old projects that are likely no longer relevant. By adding an `age_maximum_days` filter, the bot can now focus on a more specific timeframe (e.g., projects created between 5 and 180 days ago), making the alerts more actionable and reducing noise.

a-custodio committed:
feat(config): add age_maximum_days parameter

Adds the `age_maximum_days` parameter to the filters section.

This allows configuring a time window to exclude projects older than a specified number of days from the analysis. Setting the value to 0 disables the filter.

a-custodio committed:
feat(filters): add filter_younger_than function

Introduces the `filter_younger_than` function to handle project filtering based on a maximum age.

This function complements the existing `filter_older_than` and will be used to select only projects within the configured time window.

a-custodio committed:
feat(main): apply maximum age filter to project list

Updates the main execution flow to incorporate the new maximum age filter.

The application now reads the `age_maximum_days` configuration and applies the `filter_younger_than` function to the project list, ensuring that only projects within the specified age range are processed.

a-custodio committed:
docs(readme): update configuration section

Documents the new `age_maximum_days` filter parameter in the `config.yaml` details.

guilherme-hosoda-cit commented:
Pelo que entendi, esta PR está ignorando todos os projetos criados com uma idade maior que X para enviar os dados pelo Bot. Acho que a ideia seria na verdade ignorar dados de custo mais antigos que X dias. Assim, caso alguma API seja ativada em um projeto antigo aumentando os custos, ou qualquer política de custos seja alterada, não vamos ter uma surpresa pq esses valores vão continuar chegando.

a-custodio commented:
> Pelo que entendi, esta PR está ignorando todos os projetos criados com uma idade maior que X para enviar os dados pelo Bot. Acho que a ideia seria na verdade ignorar dados de custo mais antigos que X dias. Assim, caso alguma API seja ativada em um projeto antigo aumentando os custos, ou qualquer política de custos seja alterada, não vamos ter uma surpresa pq esses valores vão continuar chegando.

Perfeito, faz mais sentido. Vou providenciar a alteração!

a-custodio committed:
refactor(filters): remove project filtering based on maximum age

Removes the `filter_younger_than` function and its application.

The previous approach incorrectly filtered out projects from the analysis. This change reverts that logic to ensure all relevant projects are processed, preparing for a new implementation where the age parameter will control the cost calculation window instead.

a-custodio committed:
feat(billing): implement dynamic cost calculation window

Refactors the `query_billing_info` function to dynamically generate the BigQuery query for cost calculation.

The query is now built using the `age_maximum_days` parameter from the configuration to define a specific time window (e.g., last 45 days). To support this, the function now queries the main billing export table directly, instead of a pre-aggregated view, allowing it to filter on the `usage_start_time` column.

a-custodio committed:
docs(config): update example for dynamic billing query

Updates the `example-config.yaml` to reflect the new billing logic.

- Renames `cost_view_full_name` to `billing_export_table_name` to clarify that the source table is now required.

a-custodio committed:
docs(readme): align documentation with dynamic billing logic

Updates the README to reflect the recent changes in the cost calculation mechanism.

Key changes include:
- The "Prerequisite" section now instructs users to use the main billing export table instead of a pre-aggregated view.
- The configuration details were updated to replace `cost_view_full_name` with `billing_export_table_name`.
- The description of `age_maximum_days` was clarified to explain its new role in defining the cost calculation window.

guilherme-hosoda-cit commented on billing.py:
```diff
@@ -5,32 +5,49 @@
     client = bigquery.Client(project=BIGQUERY_CLIENT_PROJECT)
-    query_job = client.query("""
+
+    if COST_WINDOW_DAYS and COST_WINDOW_DAYS > 0:
```
NIT: 
In Python, if you run an if statement with a numeric variable, it will accept the condition as true if X > 0 and X < 0. So you can simplify this block. 

```suggestion
    if COST_WINDOW_DAYS:
```

guilherme-hosoda-cit commented on billing.py:
```diff
@@ -5,32 +5,49 @@
     for row in results:
         results_by_project[row.project_id] = {
-            'billingAccountName': row.billing_account_name,
+            'billingAccountName': row.billing_account_id,
```
Why you've remove the billing account name from the query? The field is not available in this new table?

a-custodio replied:
You're correct. The billing_account_name field isn't available in the raw billing export table we are using now. The previous VIEW had this field added manually. As a fallback, I'm using the billing_account_id for both fields since it's the most reliable identifier available from the source table.

guilherme-hosoda-cit commented on main.py:
```diff
@@ -124,7 +124,7 @@ def main():
     if DEBUG_FILTERED_BY_AGE:
         logger.debug('Aged Projects filter applied:\n%s', pformat(older_projects))
-
+        
```
NIT: typo
```suggestion

```

a-custodio committed:
refactor: apply feedback from pull request

Addresses feedback from the recent pull request to improve code quality.

- In `billing.py`, the conditional check for `COST_WINDOW_DAYS` is simplified to be more Pythonic.
- In `main.py`, an extraneous blank line was removed for better code formatting.

guilherme-hosoda-cit commented:
LGTM
